File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-0.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {number[][]}
     */
    levelOrder(root) {
        let res = []
        if(root == null){
            return res
        }
        let q = new Queue()
        q.push(root)
        while(!q.isEmpty()){
            let level = [];
            let size = q.size()
            for(let i=0; i<size; i++){
                let node = q.pop()
                if(node){
                    level.push(node.val)
                    q.push(node.left)
                    q.push(node.right)
                }
            }
            if (level.length > 0) {
                res.push(level);
            }
        }

        return res
       
    }
}

File: Data Structures & Algorithms/binary-search/submission-0.java
class Solution {
    public int search(int[] nums, int target) {
        int left = 0;
        int right = nums.length;
        while(left < right){
            int m = left + (right - left) / 2;
            if(nums[m] >= target){
                right = m;
            }
            else {
               left = m+1;
            }
        }
        return (left < nums.length && nums[left] == target) ? left : -1;
    }
}
